pkg/controller/certificates/util.go

```diff
@@ -262,6 +262,20 @@ func GenerateLocallySignedTemporaryCertificate(crt *cmapi.Certificate, pkData []
 	return b, nil
 }
 
+//RenewalTimeFunc is a custom function type for calculating renewal time of a certificate
+type RenewalTimeFunc func(time.Time, time.Time, *cmapi.Certificate) *metav1.Time
+
+// RenewalTimeWrapper returns RenewalTimeFunc implementation
+// TODO: potentially merge RenewBeforeExpiryDuration into this function and rewrite the tests accordingly
+func RenewalTimeWrapper(defaultRenewBeforeExpiryDuration time.Duration) RenewalTimeFunc {
```

Comment:
I don't really like the wrapper we have here, makes things a bit more confusing then they have to be. Can we not just have these as standard functions? What is t he purpose of `RenewalTimeFunc`?

Reply:
It is like that because I couldn't find a cleaner way to access [DefaultRenewBefore](https://github.com/jetstack/cert-manager/blob/master/pkg/apis/certmanager/v1/const.go#L32).

Having this as a standard function would mean that that value to be:

a) passed into the controller separately and then passed into the renewal time func when it's called which seems unnecessary passing it around since it's a const anyway
b) accessed directly from the renewal time func which seems like it would make a util function too 'clever'
c) accessed directly from inside controller's ProcessItem function which again seems 'unclean' since it's an external value and we might still want to mock it in tests.

Accessing it from a closure seemed similar to how we already do it in other places, i.e when accessing the clock in [trigger controller's policy func](https://github.com/jetstack/cert-manager/blob/master/pkg/controller/certificates/trigger/trigger_controller.go#L247)

Maybe I'm just overthinking it though 😄